KNIDEPLOY-4016 (#31755)
openshift/openshift-docs: modules/machine-health-checks-about.adoc

[id="machine-health-checks-about_{context}"]
= About machine health checks

Machine health checks automatically repair unhealthy machines in a particular machine pool.

To monitor machine health, create a resource to define the configuration for a controller. Set a condition to check, such as staying in the `NotReady` status for 5 minutes or displaying a permanent condition in the node-problem-detector, and a label for the set of machines to monitor.

[NOTE]
====
You cannot apply a machine health check to a machine with the master role.
====

The controller that observes a `MachineHealthCheck` resource checks for the defined condition. If a machine fails the health check, the machine is automatically deleted and a new one is created to take its place. When a machine is deleted, you see a `machine deleted` event.

To limit disruptive impact of the machine deletion, the controller drains and deletes only one node at a time. If there are more unhealthy machines than the `maxUnhealthy` threshold allows for in the targeted pool of machines, remediation stops so that manual intervention can take place.

[NOTE]
====

> Consider the timeouts carefully, depending on workloads and requirements.

Reviewer: It's probably just early, but this line doesn't scan quite right. Consider:

```suggestion
Consider the timeouts carefully, accounting for workloads and requirements.
```

* Long timeouts can result in long periods of downtime for the workload on the unhealthy machine.

> * Too short timeouts can result in a remediation loop. For example, when checking the `NotReady` status, but the machine does not

Reviewer: The first clause in you example's a fragment, and the sentence would be too long if we formatted it correctly. Please rewrite it to start with an independent clause.
